nebula_ros/src/continental/continental_ars548_decoder_ros_wrapper.cpp

```diff
@@ -0,0 +1,344 @@
+}
+
+Status ContinentalARS548DriverRosWrapper::InitializeDriver(
+  std::shared_ptr<drivers::SensorConfigurationBase> sensor_configuration)
+{
+  decoder_ptr_ = std::make_shared<drivers::continental_ars548::ContinentalARS548Decoder>(
+    std::static_pointer_cast<drivers::ContinentalARS548SensorConfiguration>(sensor_configuration));
+
+  decoder_ptr_->RegisterDetectionListCallback(std::bind(
+    &ContinentalARS548DriverRosWrapper::DetectionListCallback, this, std::placeholders::_1));
+  decoder_ptr_->RegisterObjectListCallback(
+    std::bind(&ContinentalARS548DriverRosWrapper::ObjectListCallback, this, std::placeholders::_1));
+
+  return Status::OK;
+}
+
+Status ContinentalARS548DriverRosWrapper::GetStatus()
+{
+  return wrapper_status_;
+}
+
+Status ContinentalARS548DriverRosWrapper::GetParameters(
+  drivers::ContinentalARS548SensorConfiguration & sensor_configuration)
+{
+  {
+    rcl_interfaces::msg::ParameterDescriptor descriptor;
+    descriptor.type = rcl_interfaces::msg::ParameterType::PARAMETER_STRING;
+    descriptor.read_only = true;
+    descriptor.dynamic_typing = false;
+    descriptor.additional_constraints = "";
+    this->declare_parameter<std::string>("host_ip", "255.255.255.255", descriptor);
+    sensor_configuration.host_ip = this->get_parameter("host_ip").as_string();
+  }
+  {
+    rcl_interfaces::msg::ParameterDescriptor descriptor;
+    descriptor.type = rcl_interfaces::msg::ParameterType::PARAMETER_STRING;
+    descriptor.read_only = true;
+    descriptor.dynamic_typing = false;
+    descriptor.additional_constraints = "";
+    this->declare_parameter<std::string>("sensor_ip", "192.168.1.201", descriptor);
+    sensor_configuration.sensor_ip = this->get_parameter("sensor_ip").as_string();
+  }
+  {
+    rcl_interfaces::msg::ParameterDescriptor descriptor;
+    descriptor.type = rcl_interfaces::msg::ParameterType::PARAMETER_INTEGER;
+    descriptor.read_only = true;
+    descriptor.dynamic_typing = false;
+    descriptor.additional_constraints = "";
+    this->declare_parameter<uint16_t>("data_port", 2368, descriptor);
+    sensor_configuration.data_port = this->get_parameter("data_port").as_int();
+  }
+  {
+    rcl_interfaces::msg::ParameterDescriptor descriptor;
+    descriptor.type = rcl_interfaces::msg::ParameterType::PARAMETER_STRING;
+    descriptor.read_only = true;
+    descriptor.dynamic_typing = false;
+    descriptor.additional_constraints = "";
+    this->declare_parameter<std::string>("sensor_model", "");
+    sensor_configuration.sensor_model =
+      nebula::drivers::SensorModelFromString(this->get_parameter("sensor_model").as_string());
+  }
+  {
+    rcl_interfaces::msg::ParameterDescriptor descriptor;
+    descriptor.type = rcl_interfaces::msg::ParameterType::PARAMETER_STRING;
+    descriptor.read_only = false;
+    descriptor.dynamic_typing = false;
+    descriptor.additional_constraints = "";
+    this->declare_parameter<std::string>("frame_id", "continental_ars548", descriptor);
```

Comment:
`frame_id` is declared as `continental` in `nebula_ros/src/continental/continental_ars548_hw_interface_ros_wrapper.cpp`, please make them the same everywhere.
Personally, I think changing it to just `continental` is more readable.

Reply:
Thanks !
Keeping up with nebula conventions, I changed them to `continental` in 
182c571